Newton-Gripper/Newton-Gripper.ino

```diff
@@ -167,32 +181,24 @@ void runSpeedFilter() {
     // Remove neutral PWM bias & clamp to [-HALF_RANGE, HALF_RANGE]
     int16_t pw = constrain(pulsewidth - PWM_NEUTRAL, -HALF_RANGE, HALF_RANGE);
 
-    // Reject anything inside input deadzone
     if ( pw > INPUT_DZ ) {
-    // Map valid PWM signals to (0.0 to 1.0]
-      rawvelocity = (float)(pw - INPUT_DZ)/(HALF_RANGE - INPUT_DZ)
-                    *(maxduty - minduty) + minduty;
+      rawvelocity = 1.0f;
     } else if ( pw < -INPUT_DZ ) {
-      // Map valid PWM signals to [-1.0 to 0.0)
-      rawvelocity = (float)(pw + INPUT_DZ)/(HALF_RANGE - INPUT_DZ)
-                    *(maxduty - minduty) - minduty;
+      rawvelocity = -1.0f;
     } else {
       // Stop motor if input is within input deadzone
       rawvelocity = 0.0f;
     }
-  } else {
-    // Stop motor if input is invalid
```

Comment:
wouldn't you still want to stop the motor if you get an invalid signal?

Reply:
This was one of the root causes for the "beeping". Under heavy load (near stall) the PWM measurement is unreliable due to noise on the ground. We decided that it was better to continue with the current state between invalid signals. A better error rejection method is probably possible but this has been sufficient in our testing and we haven't gotten any negative feedback from beta testers.